src/validators/validators-functions.js

```diff
@@ -0,0 +1,50 @@
+import { isFunction, isPlainObject } from '../utils/type-inference';
+import { castArray } from '../utils/array';
+
+const getValidators = (validator) => isFunction(validator) ? validator() : validator;
+
+/**
+ * check if a value is valid or not
+ * @param  {object} validators -  a config object
+ * @param  {any} value
+ * @param  {key} string
+ * @returns boolean - using the validators to check if the value is a valid value or not
+ */
+export const isValueValid = (validators, value, key) => {
+  const validatorsArray = castArray(validators);
+  const isValid = validatorsArray.some((validator) => getValidators(validator).value(value));
+  const invalidItem = validatorsArray.find((validator) => !getValidators(validator).value(value));
```

Comment:
Why is the isValid needed if you have the invalid check? can leave only the invalid check, no?

Reply:
it does not work and I understand why.... for example:  if the first validator is false the find will done (!getValidators) i need to find what causes a failure. but it does not say its in isValid because the next item can be true ... i will fix it

Reply:
@maoznir replace with a single forEach